File: lib/models/media_item.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'media_item.freezed.dart';
part 'media_item.g.dart';

@freezed
abstract class MediaItem with _$MediaItem {
  const factory MediaItem({
    required String url,
    required String fileName,
    required String extension,
    required String sourceUrl,
    int? sizeBytes,
    String? quality,
  }) = _MediaItem;

  factory MediaItem.fromJson(Map<String, dynamic> json) =>
      _$MediaItemFromJson(json);
}

File: app/src/main/java/io/github/rustnomicon/sourceinstaller/data/model/MediaItem.kt
package io.github.rustnomicon.sourceinstaller.data.model

/**
 * A single downloadable media file discovered by a scraper.
 *
 * Port of the Flutter `MediaItem` freezed model.
 */
data class MediaItem(
    val url: String,
    val fileName: String,
    val extension: String,
    val sourceUrl: String,
    val sizeBytes: Long? = null,
    val quality: String? = null,
)
